node/src/actors/session/handlers.rs

```diff
@@ -116,14 +116,33 @@ impl StreamHandler<BytesMut, Error> for Session {
                     (
                         _,
                         SessionStatus::Unconsolidated,
-                        Command::Version(Version { sender_address, .. }),
+                        Command::Version(Version {
+                            sender_address,
+                            timestamp,
+                            ..
+                        }),
                     ) => {
-                        let msgs = handshake_version(self, &sender_address);
-                        for msg in msgs {
-                            self.send_message(msg);
+                        let current_ts = get_timestamp();
+                        let received_ts = timestamp;
+
+                        if self.handshake_max_ts_diff != 0
+                            && (current_ts - received_ts).abs() > self.handshake_max_ts_diff
+                        {
+                            log::warn!(
+                                "Dropping peer because their timestamp is too big (±{} seconds",
```

Comment:
How about we show the sign of the difference, and the local timestamp:

```suggestion
                                "Dropping peer because their timestamp is different from ours ({:+} seconds), current timestamp: {}",
                                (received_ts - current_ts), current_ts,
```